### PR TITLE
Cu 86b0yptay create a challenge

### PR DESCRIPTION
**Pull request by zeckkimon**
Changes:
- Integrate with API
- Implement Backhandler
- Implement Share
- Create DatePickerModal component
- Use custom Text component
- Input validation

Not yet implemented:
- Custom component for Duration field

* **Note**: To test using your backend localhost, use your IP in the config
i.e  api_url: "http://**[your IP]**:3000/api/v1/"

### CODE REVIEW

**Comment by zeckkimon** [comment]
Task linked: [CU-86b0yptay Create a Challenge](https://app.clickup.com/t/86b0yptay)

File: src/components/forms/ChallengeConfirmationForm.tsx [hunk 1]
```diff
@@ -28,15 +45,48 @@ const ChallengeConfirmationForm = (props: any) => {
...
+            }}
+            render={({ field }) => <Text>{field.value} days</Text>}
+          />
+
+          <Stack flexDirection="row">
+            <Controller
+              control={control}
+              name="start_at"
+              rules={{
+                required: true,
```
Comment (whamester) [inline src/components/forms/ChallengeConfirmationForm.tsx]: These fields on the summary don't need the rules because they are not rendering inputs.

**Review by whamester — APPROVED** [review]

File: src/components/forms/ChallengeConfirmationForm.tsx (hunk 1 above)
Comment (zeckkimon) [reply src/components/forms/ChallengeConfirmationForm.tsx]: Thanks Wonnyo! I made changes to not use Controller and just retrieve the values. Please confirm if this is ok.

**Review by ReinhardtBotha — APPROVED** [review]
Looks great and works as expected. I just had to make sure my user had a token.

**Review by Victorportus — APPROVED** [review]

**Pull request merged**